### PR TITLE
Moved default CanBuildHere code from LUA to C#

### PR DESCRIPTION
**Pull request by mmooxy**
Slightly reduces LUA function calls.

Note: `CanPlaceHere` tag in XML (Tiles.xml) should probably match the `CanBuildHereLua` (TileTypes.cs) property and the LUA function names (`CanBuildHere_Ladder`).

### CODE REVIEW

File: Assets/Scripts/Controllers/BuildModeController.cs [hunk 1]
```diff
@@ -248,6 +248,12 @@ public bool DoesBuildJobOverlapExistingBuildJob(Tile t, string furnitureType)
     // TODO Export this kind of check to an XML/LUA file for easier modding of floor types.
     private bool CanBuildTileTypeHere(Tile t, TileType tileType)
     {
+        if (tileType.CanBuildHereLua == null)
+        {
+            // Default
```
Comment (Randshot) [inline Assets/Scripts/Controllers/BuildModeController.cs]: Add a period after a comment.

**Comment by Randshot** [comment]
Works for me :)
Code looks okay.

**Comment by Mizipzor** [comment]
This branch has conflicts that must be resolved, probably due to the huge StyleCop cleanup that just happened. Fix them and remove the `[Conflicts]` tag. :)

**Comment by vogonistic** [comment]
Merged!

**Pull request merged**